### PR TITLE
[backport] add v1.6 changelog

### PR DESCRIPTION
**Pull request by narimiran**
This will be used (linked) from the release article on our website.

Author: @timotheecour

### CODE REVIEW

File: changelogs/changelog_1_6_0.md [hunk 1]
```diff
@@ -0,0 +1,925 @@
...
+- `random.initRand(seed)` now produces non-skewed values for the first call to
+  `rand()` after initialization with a small (< 30000) seed.
+  Use `-d:nimLegacyRandomInitRand` to restore previous behavior for a transition
+  time, see PR [#17467](https://github.com/nim-lang/Nim/pull/17467).
+
+
+## `std/json`, `std/jsonutils`
+
+- `std/jsonutils` now serializes/deserializes holey enums as regular enums
+  (via `ord`) instead of as strings.
```
Comment (arnetheduck) [inline changelogs/changelog_1_6_0.md]: this means a program compiled with 1.4 no longer can interop with a program compiled with 1.6?

File: changelogs/changelog_1_6_0.md [hunk 2]
```diff
@@ -0,0 +1,925 @@
...
+  Use `-d:nimLegacyJsonutilsHoleyEnum` for a transition period.
```
Comment (arnetheduck) [inline changelogs/changelog_1_6_0.md]: per new compatibility guideline, this should then be a preview feature first

Comment (Araq) [reply changelogs/changelog_1_6_0.md]: correct.

**Comment by arnetheduck** [comment]
In general, it would be important to include a prominent list of known breaking changes - rn it's hard to find them as they're hidden between bugs and feature additions.

**Comment by dom96** [comment]
> In general, it would be important to include a prominent list of known breaking changes - rn it's hard to find them as they're hidden between bugs and feature additions.

Indeed, that is how we used to do release notes... last I heard we stopped because we don't want people to think that we are breaking things, which is just silly IMO.

**Comment by Araq** [comment]
> Indeed, that is how we used to do release notes... last I heard we stopped because we don't want people to think that we are breaking things, which is just silly IMO.

I don't think that's what I said, but where are the breaking changes for 1.6 anyway? There is one regarding jsonutils which we'll rollback...

**Comment by arnetheduck** [comment]
> want people to think

that sounds really odd though, not at all what I'd expect - ie when you break things, presumably you've thought it through and decide that the pain it causes is worth it, and can stand behind the decision.

re breaking changes, I'm not sure - there was a long list in the devel release notes - if they have all been addressed, fantastic!

I am also thinking about the chronos issue with callbacks, but that's `lent` which presumably already broke working code in 1.4?

**Comment by Araq** [comment]
> I am also thinking about the chronos issue with callbacks, but that's lent which presumably already broke working code in 1.4?

Correct.

File: changelogs/changelog_1_6_0.md [hunk 3]
```diff
@@ -0,0 +1,925 @@
...
+
+Compatibility notes:
+- `--newruntime` and `--refchecks` are deprecated,
+  use `--gc:arc`, `--gc:orc`, or `--gc:none` as appropriate instead.
+
+
+## Docgen
+
+- docgen: RST files can now use single backticks instead of double backticks and
+  correctly render in both `rst2html` (as before) as well as common tools rendering
```
Comment (a-mr) [inline changelogs/changelog_1_6_0.md]:
```suggestion
  correctly render in both `nim rst2html` (as before) as well as common tools rendering
```

File: changelogs/changelog_1_6_0.md [hunk 4]
```diff
@@ -0,0 +1,925 @@
...
+  RST directly (e.g. GitHub).
+  This is done by adding the `default-role:: code` directive inside the RST file
+  (which is now handled by `rst2html`).
```
Comment (a-mr) [inline changelogs/changelog_1_6_0.md]:
```suggestion
  (which is now handled by `nim rst2html`).
```

File: changelogs/changelog_1_6_0.md [hunk 5]
```diff
@@ -0,0 +1,925 @@
...
+- Source+Edit links now appear on top of every docgen'd page when
+  `nim doc --git.url:url ...` is given.
+- Latex doc generation is revised: output `.tex` files should be compiled
+  by `xelatex` (not by `pdflatex` as before). Now default Latex settings
+  provide support for Unicode and better avoid margin overflows.
```
Comment (a-mr) [inline changelogs/changelog_1_6_0.md:876]:
```suggestion
  provide support for Unicode and better avoid margin overflows.
  The minimum required version is TeXLive 2018 (or an equivalent MikTeX version).
```

**Comment by disruptek** [comment]
You broke CPS and my codebase at work.  Does that help?

**Comment by Araq** [comment]
> You broke CPS and my codebase at work. Does that help?

No, **why** did we break these... I cannot document breaking changes I've no clue about.

**Pull request merged**